codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/message/XMLMessageMarshallableGenerator.kt

```diff
@@ -0,0 +1,182 @@
+package software.amazon.smithy.aws.swift.codegen.message
```

Comment:
Virtually the same as `MessageMarshallableGenerator` but uses closures for serializing instead of encoder, and generates `marshal` as static function variable instead of as conformance to `MessageMarshallable` protocol. The marshallable generators & unmarshallable generators should both be refactored down the line to reduce code duplication. For now, followed the same decision that the `XMLMessageUnmarshallableGenerator` has taken.